### PR TITLE
fzf preview window support

### PR DESCRIPTION
**Pull request by Avi-D-coder**

### CODE REVIEW

**Comment by Avi-D-coder** [comment]
@liuchengxu Should be fixed now

File: autoload/vista/finder/fzf.vim [hunk 1]
```diff
@@ -56,9 +56,15 @@ function! s:Run(...) abort
   let opts = {
           \ 'source': source,
           \ 'sink': function('s:sink'),
-          \ 'options': '--prompt "'.s:cur_executive.'> "',
+          \ 'options': ['--prompt', s:cur_executive.'> '] + get(g:, 'vista_fzf_opt', []),
           \ }
 
+  if exists('g:vista_fzf_preview')
+    let preview_opts = call('fzf#vim#with_preview', g:vista_fzf_preview).options
```
Comment (liuchengxu) [inline autoload/vista/finder/fzf.vim]: I think it's more intuitive to use the option `g:vista_fzf_preview` as a String and wrap it as a List internally when calling `fzf#vim#with_preview`.

```vim
let g:vista_fzf_preview = 'right:50%'

let preview_opts = call('fzf#vim#with_preview', [g:vista_fzf_preview]).options
```

Comment (Avi-D-coder) [reply autoload/vista/finder/fzf.vim]: I made `g:vista_fzf_preview` a list to support potential use cases that need multiple arguments be passed to `fzf#vim#with_preview`. I am not sure there are use cases at this time, but applying the list as arguments to `fzf#vim#with_preview` is consistent with fzf api `function! fzf#vim#with_preview(...)`. `with_preview(...)` can gain functionality without breaking it's api so we should uphold it's api.
```
command! -bang -nargs=* Rg
  \ call fzf#vim#grep(
  \   'rg --column --line-number --no-heading --color=always --smart-case '.shellescape(<q-args>), 1,
  \   <bang>0 ? fzf#vim#with_preview('up:60%')
  \           : fzf#vim#with_preview('right:50%:hidden', '?'),
  \   <bang>0)
```

Comment (liuchengxu) [reply autoload/vista/finder/fzf.vim]: Fair enough.

File: autoload/vista/finder/fzf.vim [hunk 2]
```diff
@@ -86,11 +92,11 @@ function! s:Run(...) abort
...
 
 function! s:Highlight() abort
-  syntax match FZFVistaBracket /\[\|\]/ contained
   syntax match FZFVistaNumber /^\s*\zs\d*:\ze\w/
-  syntax match FZFVistaTag    /^\s.*\ze\[/ contains=FZFVistaNumber
-  syntax match FZFVistaScope  /^\s.*\[.*\]\ze\s/ contains=FZFVistaTag,FZFVistaBracket
-  syntax match FZFVista /^.*$/ contains=FZFVistaBracket,FZFVistaNumber,FZFVistaTag,FZFVistaScope
+  syntax match FZFVistaTag    /^[^\[]*\(\[\)\@=/ contains=FZFVistaNumber
+  syntax match FZFVistaScope  /^[^]]*]/ contains=FZFVistaTag,FZFVistaBracket
+  syntax match FZFVista /^[^│┌]*[│┌└\n]/ contains=FZFVistaBracket,FZFVistaNumber,FZFVistaTag,FZFVistaScope
```
Comment (liuchengxu) [inline autoload/vista/finder/fzf.vim]: To be honest, I never really master these opaque patterns :(. I don't fully understand `/^[^│┌]*[│┌└\n]/`, but it does work great.

Comment (Avi-D-coder) [reply autoload/vista/finder/fzf.vim]: Looking at that with fresh eyes it should be `^[^│┌└]*` (fixed)

The highlighter still is not optimal (has theoretical bugs), I'll send a different pull request when I get a chance.

**Comment by liuchengxu** [comment]
Let's merge this and improve it later, thank you for the nice work!

**Pull request merged**